presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -1027,6 +1038,139 @@ core::LocalPartitionNode::Type toLocalExchangeType(
 }
+
+std::vector<std::shared_ptr<const connector::hive::HiveColumnHandle>>
+toHiveColumns(
+    const protocol::List<protocol::HiveColumnHandle>& inputColumns,
+    bool& hasPartitionColumn) {
+  hasPartitionColumn = false;
+  std::vector<std::shared_ptr<const connector::hive::HiveColumnHandle>>
+      hiveColumns;
+  hiveColumns.reserve(inputColumns.size());
+  for (const auto& columnHandle : inputColumns) {
+    hasPartitionColumn |=
+        columnHandle.columnType == protocol::ColumnType::PARTITION_KEY;
+    hiveColumns.emplace_back(
+        std::dynamic_pointer_cast<connector::hive::HiveColumnHandle>(
+            toColumnHandle(&columnHandle)));
+  }
+  return hiveColumns;
+}
+
+HiveBucketProperty::Kind toHiveBucketPropertyKind(
+    protocol::BucketFunctionType bucketFuncType) {
+  switch (bucketFuncType) {
+    case protocol::BucketFunctionType::PRESTO_NATIVE:
+      return HiveBucketProperty::Kind::kPrestoNative;
+    case protocol::BucketFunctionType::HIVE_COMPATIBLE:
+      return HiveBucketProperty::Kind::kHiveCompatible;
+    default:
+      VELOX_USER_FAIL(
+          "Unknown hive bucket function: {}", toJsonString(bucketFuncType));
+  }
+}
+
+std::vector<std::string> fromStringList(
+    const protocol::List<protocol::String>& stringList) {
+  std::vector<std::string> stringVector;
+  stringVector.reserve(stringList.size());
+  for (const auto& string : stringList) {
+    stringVector.push_back(string);
+  }
+  return stringVector;
+}
+
+core::SortOrder toSortOrder(protocol::Order order) {
+  switch (order) {
+    case protocol::Order::ASCENDING:
+      return core::SortOrder(true, true);
+    case protocol::Order::DESCENDING:
+      return core::SortOrder(false, false);
+    default:
+      VELOX_USER_FAIL("Unknown sort order: {}", toJsonString(order));
+  }
+}
+
+std::shared_ptr<HiveSortingColumn> toHiveSortingColumn(
+    const protocol::SortingColumn& sortingColumn) {
+  return std::make_shared<HiveSortingColumn>(
+      sortingColumn.columnName, toSortOrder(sortingColumn.order));
+}
+
+std::vector<std::shared_ptr<const HiveSortingColumn>> toHiveSortingColumns(
+    const protocol::List<protocol::SortingColumn>& sortedBy) {
+  std::vector<std::shared_ptr<const HiveSortingColumn>> sortingColumns;
+  sortingColumns.reserve(sortedBy.size());
+  for (const auto& sortingColumn : sortedBy) {
+    sortingColumns.push_back(toHiveSortingColumn(sortingColumn));
+  }
+  return sortingColumns;
+}
+
+std::shared_ptr<HiveBucketProperty> toHiveBucketProperty(
+    const std::vector<std::shared_ptr<const connector::hive::HiveColumnHandle>>&
+        inputColumns,
+    const std::shared_ptr<protocol::HiveBucketProperty>& bucketProperty) {
+  if (bucketProperty == nullptr) {
+    return nullptr;
+  }
+  VELOX_USER_CHECK(
+      bucketProperty->sortedBy.empty(),
+      "Bucketed sorted table is not supported: {}",
+      toJsonString(*bucketProperty));
+
+  const int32_t bucketCount = bucketProperty->bucketCount;
+  if (bucketCount <= 0) {
```

Comment:
Is this possible? Should this be a check?

Reply:
Not possible but there is a same check on Java side. Change it to VELOX_CHECK. Thanks!